flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTableImpl.java

```diff
@@ -118,7 +129,8 @@ public TaskSlotTableImpl(
             final ResourceProfile defaultSlotResourceProfile,
             final int memoryPageSize,
             final TimerService<AllocationID> timerService,
-            final Executor memoryVerificationExecutor) {
+            final Executor memoryVerificationExecutor,
+            final String workingDirectory) {
```

Comment:
I would pass in a `File` because it already restricts the value space compared to `String`.